### PR TITLE
update default parameters as suggested in #1606

### PR DESCRIPTION
**Pull request by yoshitomo-matsubara**
As discussed in #1606 the default hyperparameters in [detection/train.py](https://github.com/pytorch/vision/blob/master/references/detection/train.py) are different from those used to train Faster/Mask/Keypoint R-CNNs.

Following a PR suggestion from @fmassa, I updated the default hyperparameters, and added some comments so that we can have a better idea as we train them by ourselves e.g., using different datasets, models, etc.

Thank you!

### CODE REVIEW

**Comment by yoshitomo-matsubara** [comment]
Sure, I sent a follow-up PR #1612 :)
